src/components/CIPipelineN/ConditionDataTable/ConditionDataTable.component.tsx

```diff
@@ -0,0 +1,259 @@
+import { useContext, useMemo } from 'react'
+
+import {
+    Button,
+    ButtonVariantType,
+    ConditionDataTableHeaderKeys,
+    ConditionDetails,
+    ConditionType,
+    DynamicDataTable,
+    Icon,
+    PluginType,
+    RadioGroup,
+    RadioGroupItem,
+} from '@devtron-labs/devtron-fe-common-lib'
+
+import { ConditionContainerType } from '@Components/ciPipeline/types'
+import { pipelineContext } from '@Components/workflowEditor/workflowEditor'
+
+import { CONDITION_DATA_TABLE_OPERATOR_OPTIONS } from './constants'
+import {
+    ConditionDataTableActionType,
+    ConditionDataTableCustomState,
+    ConditionDataTableProps,
+    ConditionDataTableType,
+    HandleRowUpdateActionProps,
+} from './types'
+import {
+    convertConditionDataTableToFormData,
+    getConditionDataTableCellValidateState,
+    getConditionDataTableHeaders,
+    getConditionDataTableInitialCellError,
+    getConditionDataTableRowEmptyValidationState,
+    getConditionDataTableRows,
+} from './utils'
+
+export const ConditionDataTable = ({ type, conditionType, handleConditionTypeChange }: ConditionDataTableProps) => {
+    // CONTEXTS
+    const {
+        activeStageName,
+        selectedTaskIndex,
+        formData,
+        formDataErrorObj,
+        setFormData,
+        setFormDataErrorObj,
+        validateTask,
+    } = useContext(pipelineContext)
+
+    // CONSTANTS
+    const currentStepTypeVariable =
+        formData[activeStageName].steps[selectedTaskIndex].stepType === PluginType.INLINE
+            ? 'inlineStepDetail'
+            : 'pluginRefStepDetail'
+
+    const conditionDetails: ConditionDetails[] =
+        formData[activeStageName].steps[selectedTaskIndex][currentStepTypeVariable]?.conditionDetails
+
+    const ioVariables =
+        formData[activeStageName].steps[selectedTaskIndex][currentStepTypeVariable][
+            type === ConditionContainerType.PASS_FAILURE ? 'outputVariables' : 'inputVariables'
+        ]
+
+    const ioVariablesError =
+        formDataErrorObj[activeStageName].steps[selectedTaskIndex][currentStepTypeVariable]?.conditionDetails || {}
+
+    const headers = getConditionDataTableHeaders(conditionType)
+
+    // TABLE ROWS
+    const rows = useMemo<ConditionDataTableType['rows']>(
```

Comment:
This useMemo won't be optimizing anything. We should remove this. as the dependencies will be recreated on every render